recipes/repo.rb

```diff
@@ -284,6 +284,102 @@
   end
 end
 
+# RPM repository setup
+apt_repository 'createrepo-agent-ppa' do
+  uri 'ppa:osrf/createrepo-agent'
+end
+
+package 'createrepo-agent'
+
+package 'createrepo-c'
+
+package 'rpm'
+
+package 'socat'
```

Comment:
What introduces the socat and rpm dependencies? Are we using those tools directly or are they dependencies of optional features in createrepo-agent that we're using?

Reply:
An aside on the formatting, I don't feel strongly about any specific option including leaving it as is, but I don't see a need to add empty lines between one-line resource declarations. It's also possible to use a short loop to express installation for a collection of related packages.

```suggestion
package 'createrepo-agent'
package 'createrepo-c'
package 'rpm'
package 'socat'
```

```suggestion
%w[createrepo-agent createrepo-c rpm socat].each do |pkg|
  package pkg
end
```

Reply:
The `rpm` tool is needed to sign the packages themselves, which we're not quite doing yet, but I'd like to start doing instead of metadata signing.

The `socat` tool isn't strictly needed, but it can be used to essentially open a shell to `createrepo-agent` to debug problems and perform manual modifications.

Reply:
> `socat` ... can be used to debug problems and perform manual modifications.

Sounds good. I think a comment to that effect would be welcome but hardly mandatory. In the context of security review socat is one of those tools that's sharp enough to cut yourself on so we'd want sound justification for keeping it on a production host (which we have).